src/k2/module/k23si/client/k23si_client.cpp

```diff
@@ -241,9 +241,16 @@ seastar::future<EndResult> K2TxnHandle::end(bool shouldCommit) {
             K2LOG_W(log::skvclient, "Tried to commit a failed RO transaction, mtr={}", _mtr);
             return seastar::make_ready_future<EndResult>(EndResult(_failed_status));
         }
-
+        auto time_spent = Clock::now() - _start_time;
+        auto minTransTime = _client->getTSOErrorbound();
+        if (time_spent < minTransTime) {
+            auto sleep = minTransTime - time_spent;
+            return seastar::sleep(sleep).then([] () {
+                return seastar::make_ready_future<EndResult>(EndResult(Statuses::S200_OK("default end result")));
+            });
+        }
         reporter.report();
-        _client->_txnDuration.add(k2::Clock::now() - _startTime);
+        _client->_txnDuration.add(time_spent);
```

Comment:
Need to add to the txnDuration and report() in the sleep path above too.

Reply:
Yeah, I tried to add the report and Prometheus add before but it gives assertion errors in both tpcc test and ycsb test. Will look into it tomorrow.

Reply:
hmm it seems like for read only transaction that is required to sleep for the minimum transaction time, the `Clock::now()-_start_time` gives a value < 0 and fails the assertion in Prometheus.